Frontend/ui-customer/src/Main.purs

```diff
@@ -85,10 +85,16 @@ onConnectivityEvent triggertype = do
                   "LOCATION_DISABLED" -> do
                     modifyScreenState $ PermissionScreenStateType (\permissionScreen -> permissionScreen {stage = LOCATION_DISABLED})
                     Flow.permissionScreenFlow
-                  "INTERNET_ACTION" -> do
-                    modifyScreenState $ PermissionScreenStateType (\permissionScreen -> permissionScreen {stage = INTERNET_ACTION})
-                    Flow.permissionScreenFlow
+                  "INTERNET_DISCONNECTED" -> do
+                    if(isLocalStageOn RideAccepted || isLocalStageOn RideStarted) then do
```

Comment:
Instead of creating new flow try to handle it based on callbacks. creating new flow will delete the data which can be shown in UI.

Callbacks approach can be done via hotpush and backward compatibility can be maintained.